ScoreSaber/UI/Elements/Profile/ProfileDetailView.cs

```diff
@@ -366,18 +370,37 @@ private void OpenMap(string hash, int diff) {
             }
 
             diff = (diff - 1) / 2;
-            BeatmapKey key = level.GetBeatmapKeys().FirstOrDefault(x => diff == (int)x.difficulty);
+            BeatmapKey key = level.GetBeatmapKeys().LastOrDefault(x => diff == (int)x.difficulty);
             if (key == null) {
                 Plugin.Log.Error($"Difficulty {diff} not found for level {hash}");
                 return;
             }
             LevelSelectionFlowCoordinator.State state = new LevelSelectionFlowCoordinator.State(SelectLevelCategoryViewController.LevelCategory.All,
-                                                                                                SongCore.Loader.CustomLevelsPack,
+                                                                                                _beatmapLevelsModel.GetLevelPackForLevelId(hash),
                                                                                                 in key,
                                                                                                 level);
-            //Plugin.Log.Notice($"Opening level {hash} with base game diff {diff}");
-            
-            
+
+            var favouriteids = _soloFreePlayFlowCoordinator.levelSelectionNavigationController._levelCollectionNavigationController._levelCollectionViewController._levelCollectionTableView._favoriteLevelIds;
+            var levelcollection = _soloFreePlayFlowCoordinator.levelSelectionNavigationController._levelCollectionNavigationController._levelCollectionViewController;
+            levelcollection._levelCollectionTableView.SetData(new List<BeatmapLevel>() { level }, favouriteids, false, false);
+            levelcollection.SelectLevel(level);
+        }
+
+        private void OpenOSTDLCMap(string id, int diff) {
+
+            BeatmapLevel level = _beatmapLevelsModel.GetBeatmapLevel(id);
+
+            diff = (diff - 1) / 2;
+            BeatmapKey key = level.GetBeatmapKeys().LastOrDefault(x => diff == (int)x.difficulty);
+            if (key == null) {
+                Plugin.Log.Error($"Difficulty {diff} not found for level {id}");
+                return;
+            }
```

Comment:
This code should not be duplicated, use separate functions for shared logic.

Reply:
fixed